pennylane_sf/remote.py

```diff
@@ -150,11 +154,11 @@ def probability(self, wires=None):  # pylint: disable=missing-function-docstring
             all_probs = OrderedDict((tuple(k), v) for k, v in zip(ind, all_probs))
             return all_probs
 
-        all_wires = np.arange(self.num_wires)
-        wires_to_trace_out = np.setdiff1d(all_wires, wires)
+        wires_to_trace_out = Wires.unique_wires([self.wires, wires])
+        device_wires_to_trace_out = self.map_wires(wires_to_trace_out)
 
-        if wires_to_trace_out.size > 0:
-            all_probs = np.sum(all_probs, axis=tuple(wires_to_trace_out))
+        if len(device_wires_to_trace_out) > 0:
+            all_probs = np.sum(all_probs, axis=device_wires_to_trace_out.labels)
```

Comment:
I think the size comparison was from a time where wires could be nested. This is now impossible.

Reply:
Would be worth adding a test case on unique labels to see that this one comes around.

Reply:
I'm wondering, isn't that case tested in `test_mocked_probability`? Since `probs` is called with 2 out of 8 wires? Or do I understand your comment wrong?

Reply:
Oh, sorry for the wording! Was thinking of a small example where the device has a wires labeling defined, e.g. by defining ["a", "c", "d","e","f","g","h","i"] as wires. Then 2 of them could be passed to `qml.probs`. In `test_mocked_probability` the `wires` variable is not actually used when the device is being defined. :confused:

Reply:
Something like this test?

```    python
    
        shots = 10
        wires = ["a", "c", "d", "b", "f", "e", "h", "g"]

        monkeypatch.setattr("strawberryfields.RemoteEngine", MockEngine)
        device = qml.device("strawberryfields.remote", wires=wires, backend="X8", shots=shots)

        @qml.qnode(device)
        def circuit():
            return qml.probs(wires=["a"])

        expected = np.array([0., 0.4, 0.2, 0.1, 0.3])

        assert np.allclose(circuit(), expected)
```

I must say I don't really understand the expected output given ``MOCK_DEVICE``s samples

```
MOCK_SAMPLES = np.array(
    [
        [3, 4, 2, 3, 4, 3, 1, 0],
        [4, 3, 3, 2, 0, 3, 1, 4],
        [2, 1, 3, 3, 3, 2, 2, 4],
        [4, 1, 4, 4, 2, 3, 3, 0],
        [4, 2, 3, 3, 3, 0, 0, 4],
        [1, 2, 4, 4, 2, 0, 0, 4],
        [2, 3, 1, 2, 1, 0, 4, 1],
        [1, 2, 0, 1, 2, 3, 3, 0],
        [1, 2, 4, 0, 0, 4, 2, 4],
        [1, 0, 1, 1, 1, 3, 1, 0],
    ]
)
```

At the moment, the expected output is just retrieved from running the test and copying the actual result - but of course I want to understand it before merging...

Reply:
Do you see why the probability of the first wire is `[0., 0.4, 0.2, 0.1, 0.3]`? What are the 5 numbers here? Is there an implicit cutoff of 5?

Reply:
Feel free to merge if this added test makes sense to you, otherwise I'll get back to it tomorrow!